packages/ui/react-ui/src/WalletModal.tsx

```diff
@@ -50,7 +51,7 @@ export const WalletModal: FC<WalletModalProps> = ({ className = '', container =
                   wallets.find((wallet: { readyState: any }) => wallet.readyState === WalletReadyState.Loadable) ||
                   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                   otherWallets[0]!;
-    }, [installedWallets, wallets, otherWallets]);
+    }, [installedWallets, wallets, loadableWallets, notDetectedWallets]);
```

Comment:
We should get rid of the special handling of Phantom and Torus while we're here making this change.